src/big/big_float.cr

```diff
@@ -77,11 +77,11 @@ struct BigFloat < Float
     new(mpf)
   end
 
-  def self.default_precision
+  def self.default_precision : UInt64
```

Comment:
The LibGMP method actually returns a `LibGMP::BitcntT` => `LibGMP::UI` => `LibC::ULong`, so the method returns a target dependent integer.

I think this outlines of the tool: it infers types for a specific target. It also exposes an issue in the stdlib: it leaks target dependent types.

Reply:
Any recommendations on what this line should  be? I can remove the return statement here, and agreed about the rest of your statement about leaking target dependent types.

Reply:
Ah, I see the failed build as a consequence of this change. I'll remove the return type at least.

Reply:
Instead of looking at the type of an expression in the current program, the typer tool could also apply some more generic type inference based on restrictions of the source/sink of these expressions.
In this case, it's pretty obvious that the return type of this method should be equal to the type restriction of `LibGMP.mpf_get_default_prec`, which is `LibGMP::BitcntT`.

Going this way ensures using the correct, target-independent aliases.

Reply:
Makes sense - probably more work than is justified for this kind of edge case, but doing more inspection up and down the call-chain has been something I've been curious to do (for additional static code analysis type things).